src/modules/simulator/simulator.h

```diff
@@ -236,10 +234,24 @@ class Simulator : public ModuleParams
 
 	hrt_abstime _last_sim_timestamp{0};
 	hrt_abstime _last_sitl_timestamp{0};
+	hrt_abstime _last_battery_timestamp{0};
 
-	// Lib used to do the battery calculations.
-	Battery _battery {};
-	battery_status_s _battery_status{};
+	// Because the simulator doesn't actually care about the real values of these, just stick
+	// in some good defaults.
+	// This is an anonymous class, because BatteryBase is abstract and can't directly be instantiated.
+	class : public BatteryBase
```

Comment:
I have never seen this before :astonished:. I guess it works?

Reply:
Yep! It's the same way you can declare an anonymous struct type:
```
struct {
  int a;
  float b;
} anonymous_struct;
anonymous_struct.a = 1;
```

Reply:
Ah!